pulser/sequence.py

```diff
@@ -21,23 +33,20 @@
 from itertools import chain
 import json
 from typing import Any, cast, NamedTuple, Optional, Tuple, Union
-from typing_extensions import Literal
-
 import warnings
 
-import numpy as np
-from numpy.typing import ArrayLike
+from sys import version_info
+if version_info[:2] == (3, 7):  # pragma: no cover
+    try:
+        from typing_extensions import Literal, get_args
+    except ImportError:
+        raise ImportError(
+            "Using pulser with Python version 3.7 requires the"
+            " `typing_extensions` module. Install it by running"
+            " `pip install typing-extensions`.")
+else:
+    from typing import Literal, get_args  # type: ignore
```

Comment:
Did you mean # pragma: no cover?

Reply:
?

Reply:
You wrote `# type: ignore`, was `mypy` giving you an error? This line needs to be ignored in the `coverage`, it's what's giving you the failed CI test. So it seems you wrote `# type: ignore` instead of `# pragma: no cover`

Reply:
`mypy` does complain without the `# type: ignore`: `pulser\sequence.py:48: error: Incompatible import of "Literal" (imported name has type "typing._SpecialForm", local name has type "typing_extensions._SpecialForm")`.

`# pragma: no cover` is on the same line as the `if` as advised in docs. It works fine when i `pytest` locally but not in CI.

Those are separate issues.

Reply:
Well... The `# pragma: no cover` should be on the `else` too...